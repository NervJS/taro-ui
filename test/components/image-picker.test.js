import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtImagePicker from '../../.temp/components/image-picker/index'

describe('AtImagePicker Snap', () => {
  const files = [
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121'
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
      id: '2122'
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121'
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121'
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121'
    }
  ]
  it('render initial AtImagePicker', () => {
    const component = renderToString(<AtImagePicker isTest />)
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props className', () => {
    const component = renderToString(<AtImagePicker isTest className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props customStyle', () => {
    const component = renderToString(
      <AtImagePicker isTest customStyle='color: red' />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props files', () => {
    const component = renderToString(<AtImagePicker isTest files={files} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props length', () => {
    const component = renderToString(
      <AtImagePicker isTest length={5} files={files} />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props mode', () => {
    const component = renderToString(
      <AtImagePicker isTest mode='top' files={files} />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props showAddBtn', () => {
    const component = renderToString(
      <AtImagePicker isTest showAddBtn={false} files={files} />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtImagePicker -- props multiple', () => {
    const component = renderToString(<AtImagePicker isTest multiple />)
    expect(component).toMatchSnapshot()
  })
})
