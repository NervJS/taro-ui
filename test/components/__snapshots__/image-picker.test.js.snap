// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtImagePicker Snap render AtImagePicker -- props className 1`] = `"<div class=\\"at-image-picker test\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render AtImagePicker -- props customStyle 1`] = `"<div style=\\"color: red\\" class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render AtImagePicker -- props files 1`] = `"<div class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div></div><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render AtImagePicker -- props length 1`] = `"<div class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div></div><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render AtImagePicker -- props mode 1`] = `"<div class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-top\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-top\\" src=\\"https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-top\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-top\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div></div><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-top\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render AtImagePicker -- props multiple 1`] = `"<div class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render AtImagePicker -- props showAddBtn 1`] = `"<div class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div></div><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item\\"><div class=\\"at-image-picker__remove-btn\\"></div><div class=\\"taro-img at-image-picker__preview-img\\"><img class=\\"taro-img__mode-aspectfill taro-img__mode-aspectfill--width\\" src=\\"https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg\\"/></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;

exports[`AtImagePicker Snap render initial AtImagePicker 1`] = `"<div class=\\"at-image-picker\\"><div class=\\"at-image-picker__flex-box\\"><div class=\\"at-image-picker__flex-item\\"><div class=\\"at-image-picker__item at-image-picker__choose-btn\\"><div class=\\"add-bar\\"></div><div class=\\"add-bar\\"></div></div></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div><div class=\\"at-image-picker__flex-item\\"></div></div></div>"`;
